include/bluetooth/mesh/sensor.h

```diff
@@ -258,7 +264,7 @@ struct bt_mesh_sensor_setting {
 	 *
 	 *  @return 0 on success, or (negative) error code otherwise.
 	 */
-	int (*set)(
+	int (*set)(struct bt_mesh_sensor_srv *srv,
```

Comment:
Well, this could be done, but I don't like it because it breaks the uniformity of model APIs. All other models that have get/set API calls have the `srv` context. Secondly, `bt_mesh_sensor`  is always used in array form. Inserting srv context pointer there will needlessly create 'n' number of copies of the server pointer and it also does not solve the original problem where exact same sensor array is used for four different instances of the sensor server. 

If argument list size is okay to be expanded, I think this is the only reasonable option. With respect to API breakage, i would categorize this as low impact change since it does not require adding any new behavior or handling in applications.

Reply:
> the original problem where exact same sensor array is used for four different instances of the sensor server.

I'm sorry, I didn't realize this is what we're trying to do here. This sounds even less like a valid usecase to me. Is there a valid reason to expose the exact same sensors in multiple sensor servers? 

> Also, I don't like it because it breaks the uniformity of model APIs. All other models that have get/set API calls have the srv context. But that is a minor argument though.

Please keep in mind that this is *not* a model API.

>  i would categorize this as low impact change since it does not require adding any new behavior or handling in applications.

It's a hard compilation error though, that affects every user of the sensor model subsystem. It doesn't matter that they're using the new parameter, they have to change every sensor instance in their application.

If there's a valid reason why anyone would expose the exact same sensor instance in multiple servers, I'm okay with this solution. I can't see any immediate reasons, other than "we mistakenly did this from the beginning, and now we want the API to bend to our mistake instead of fixing it", which I don't think is valid. Happy to hear counter arguments though.

Reply:
Yes, the use case is that one sensor set comprises, say 4 sensors. Each mesh device supports 4 instances of this "set" on 4 (or n) different sensor model instances. So a device as a whole represents the sensor array each set in the array does the same kind of measurement. The end-user can use this array for various purposes. 

Having the same sensor array optimizes the number of individual callbacks you need and application logic to resolve these instances.